### PR TITLE
GH-2827 -  DataKeyStore

### PR DESCRIPTION
**Pull request by biafra23**
Handles #2827

Changes proposed in this pull request:
- Add data KeyStore for delegate keys
- 
-

### CODE REVIEW

File: Multisig/Features/Security/KeychainItem.swift [hunk 1]
```diff
@@ -31,23 +55,23 @@ enum KeychainItem {
...
             if let context = LAContext(password: password) {
                 result[kSecUseAuthenticationContext] = context
             }
-        case let .ecPubKey(tag, _):
+        case let .ecPubKey(tag, service, _):
             result = [
                 kSecClass: kSecClassKey,
                 kSecAttrKeyClass: kSecAttrKeyClassPublic,
-                kSecAttrApplicationTag: tag.data(using: .utf8)!,
+                kSecAttrApplicationTag: (tag + ":" + service).data(using: .utf8)!,
```
Comment (elgatovital) [inline Multisig/Features/Security/KeychainItem.swift]: could we extract this string pattern (tag + ":" + service) to a constant and reuse it?

File: MultisigTests/Features/Security/KeychainItemStoreTests.swift [hunk 2]
```diff
@@ -20,61 +21,61 @@ class KeychainItemStoreTests: XCTestCase {
...
         //Then
-        XCTAssertTrue(try kciStore.find(KeychainItem.ecPubKey()) == nil, "Delete item failed")
+        XCTAssertTrue(try kciStore.find(kciFactory.ecPubKey()) == nil, "Delete item failed")
     }
 
     func testStoreAndRetrievePasscode() throws {
         // Given
         let randomString = UUID().uuidString
-        let passCodeData = try kciStore.find(KeychainItem.generic(id: SensitiveStore.derivedPasswordTag, service: ProtectionClass.sensitive.service())) as! Data?
+        let passCodeData = try kciStore.find(KeychainItem.generic(id: ProtectedKeyStore.derivedPasswordTag, service: ProtectionClass.sensitive.service())) as! Data?
```
Comment (elgatovital) [inline MultisigTests/Features/Security/KeychainItemStoreTests.swift]: it is confusing that ProtectedKeyStore items are used in this test case and not in ProtectedKeyStoreTests.

**Review by DmitryBespalov — APPROVED** [review]
Approve provided small refactor (extract 'tag') is done

File: Multisig/Features/Security/KeychainItem.swift (hunk 1 above)
Comment (biafra23) [reply Multisig/Features/Security/KeychainItem.swift]: I can extract the whole thing to a function and/or the delimiter (":") to a constant. But `tag` and `service` are local variables here.

**Review by elgatovital — APPROVED** [review]

**Pull request merged**